src/server.cc

```diff
@@ -248,20 +244,26 @@ void Server::shutdown()
 {
     int ret = 0; // return code
 
-    /*
-    ContentManager::getInstance()->shutdown();
-    UpdateManager::getInstance()->shutdown();
-    Storage::getInstance()->shutdown();
-    */
+#if 0 // enable later
+#ifdef HAVE_LASTFMLIB
+    last_fm.shutdown();
+    last_fm = nullptr;
+#endif
+    content->shutdown();
+    content = nullptr;
+    session_manager = nullptr;
+    update_manager->shutdown();
+    update_manager = nullptr;
+#endif
 
-    ConfigManager::getInstance()->emptyBookmark();
+    config->emptyBookmark();
     server_shutdown_flag = true;
 
     log_debug("Server shutting down\n");
 
     ret = UpnpUnRegisterRootDevice(deviceHandle);
     if (ret != UPNP_E_SUCCESS) {
-        throw _UpnpException(ret, "upnp_cleanup: UpnpUnRegisterRootDevice failed");
+        log_error("upnp_cleanup: UpnpUnRegisterRootDevice failed: %i", ret);
     }
```

Comment:
Why no exception any more?

Reply:
we are in the shutdown phase, i think its better to execute the following code in that function too instead of skipping...

Reply:
Yep, good spot, missed the context.